test/differentials/thunks.jl

```diff
@@ -1,6 +1,12 @@
 @testset "Thunk" begin
     @test @thunk(3) isa Thunk
 
+    @testset "==" begin
+        @test @thunk(3.2) == InplaceableThunk(@thunk(3.2), x -> x + 3.2)
```

Comment:
Do we want to have this? Or should we separate between Thunks and InplaceableThnks?

Reply:
`(@thunk(3.2))` == `3.2`

`Inplacable(@thunk(3.2), _->error("this shouldn;t be hit"))` == `3.2`

Thus by transitivity thye must be equal to each other